src/virtual-source-buffer.js

```diff
@@ -263,6 +263,12 @@ export default class VirtualSourceBuffer extends videojs.EventTarget {
    * @param {Event} event the done event from the transmuxer
    */
   done_(event) {
+    // Don't process and append data if the mediaSource is closed
+    if (this.mediaSource_.readyState === 'closed') {
```

Comment:
How did we get to the point where we're trying to append data even though the media source is closed?